test/health_check_test_case.py

```diff
@@ -176,16 +215,19 @@ def _test(self, endpoint_states: Mapping[str, bool], lambdas_up: bool, path: str
         with ResponsesHelper() as helper:
             helper.add_passthru(self.base_url)
             self._mock_other_lambdas(helper, lambdas_up)
-            self._mock_service_endpoints(helper, endpoint_states)
-            with mock.patch.dict(os.environ, AWS_DEFAULT_REGION='us-east-1'):
+            with self._mock_service_endpoints(helper, endpoint_states):
                 return requests.get(self.base_url + path)
 
+    @contextmanager
     def _mock_service_endpoints(self, helper: ResponsesHelper, endpoint_states: Mapping[str, bool]) -> None:
-        for endpoint, endpoint_up in endpoint_states.items():
-            helper.add(responses.Response(method='HEAD',
-                                          url=config.service_endpoint() + endpoint,
-                                          status=200 if endpoint_up else 503,
-                                          json={}))
+        with mock.patch.dict(os.environ, AWS_DEFAULT_REGION='us-east-1'):
```

Comment:
This could move down to right before the yield. Then you wouldn't have to indent the loop.